### DISCUSSION TITLE
JPEG XL support

### DISCUSSION THREAD

**Discussion by Weissrolf** [pixijs/pixijs#8108] [Ideas]
Hello. With the JPEG XL format being frozen, a free decoding library being available and browser support slowly trickling in, are there plans for JPEG XL support in PixiJS?

**Comment by Weissrolf** [comment]
Someone claimed that Pixi already supports Jpeg XL, is that true?

**Reply by bigtimebuddy** [reply]
Pixi does not support JPEG XL. Generally, Pixi does not do bleeding-edge web features until there is widespread adoption. Currently, many browser still only support JPEG XL with feature flags.

The best opportunity to use this new format is if the community develops a loader plugin to decode in software. The core team has no plans to develop such a plugin yet.

**Comment by Weissrolf** [comment]
Thanks for the information.

**Comment by OmegaRogue** [comment]
Any update on this? In 3 days Safari will have Jpegxl support in stable

**Reply by SuperSodaSea** [reply]
There are still very little browsers that support JPEG XL ([< 1%](https://caniuse.com/jpegxl)). Maybe we can add a method like `PIXI.loadTextures.addFormat({ mime: 'image/jxl', extension: '.jxl' })`, so that users can extend the format that `@pixi/assets` can load, if they want to load some new format.

**Reply by OmegaRogue** [reply]
If a method like that is added, how would i go about adding support for the format myself? I'm asking this as someone completely outside the PixiJS project, tho I work on a third party client for FoundryVTT, which does use PixiJS, and PixiJS supporting JPEGXL is one of the requirements for everything to support JPEGXL

**Reply by OmegaRogue** [reply]
> There are still very little browsers that support JPEG XL ([< 1%](https://caniuse.com/jpegxl)). Maybe we can add a method like `PIXI.loadTextures.addFormat({ mime: 'image/jxl', extension: '.jxl' })`, so that users can extend the format that `@pixi/assets` can load, if they want to load some new format.

Also, that percentage will definitely rise once safari 17 is released in a few days
